Feedstock transport costs (carbon, lime, iron ore pellet)

jaredthomas68 opened the pull request:
<!--
IMPORTANT NOTES

1. Use GH flavored markdown when writing your description:
   https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

2. If all boxes in the PR Checklist cannot be checked, this PR should be marked as a draft.

3. DO NOT DELTE ANYTHING FROM THIS TEMPLATE. If a section does not apply to you, simply write
   "N/A" in the description.

4. Code snippets to highlight new, modified, or problematic functionality are highly encouraged,
   though not required. Be sure to use proper code higlighting as demonstrated below.

   ```python
    def a_func():
        return 1

    a = 1
    b = a_func()
    print(a + b)
    ```
-->

<!--The title should clearly define your contribution succinctly.-->
# Feedstock transport costs (carbon, lime, iron ore pellet)

<!-- Describe your feature here. Please include any code snippets or examples in this section. -->
Added feedstock transport costs (carbon, lime, iron ore pellet). The feedstock transport impacts the plant installation costs, leveled cost of steel, and the cost of feedstocks in the financial model in profast.

## PR Checklist

<!--Tick these boxes if they are complete, or format them as "[x]" for the markdown to render. -->
- [x] `CHANGELOG.md` has been updated to describe the changes made in this PR
- [x] Documentation
  - [x] Docstrings are up-to-date
  - [-] Related `docs/` files are up-to-date, or added when necessary
  - [-] Documentation has been rebuilt successfully
  - [-] Examples have been updated
- [x] Tests pass (If not, and this is expected, please elaborate in the tests section)
- [x] PR description thoroughly describes the new feature, bug fix, etc.

## Related issues

<!--If one exists, link to a related GitHub Issue.-->


## Impacted areas of the software

<!--
Replace the below example with any added or modified files, and briefly describe what has been changed or added, and why.
-->
- `greenheart/simulation/technologies/steel/steel.py`
  - `class Feedstocks`: added feedstock transportation costs for iron ore pellets, lime, and carbon with "0" default values

## Additional supporting information
<!--Add any other context about the problem here.-->


## Test results, if applicable

<!--
Add the results from unit tests and regression tests here along with justification for any
failing test cases.
-->


<!--
__ For NREL use __
Release checklist:
- [ ] Update the version in greenheart/__init__.py
- [ ] Verify docs builds correctly
- [ ] Create a tag on the main branch in the NREL/GreenHEART repository and push
- [ ] Ensure the Test PyPI build is successful
- [ ] Create a release on the main branch
-->

mariya-koleva reviewed (commented):
All looks good. One minor thing: the transportation costs (lines 66, 68 and 71) are in $/tonne of feedstock. These are added to the unit costs which are also expressed in $/tonne. Also, why are the values zeroed?

jaredthomas68 commented:
> All looks good. (1) One minor thing: the transportation costs (lines 66, 68 and 71) are in per-tonne of feedstock. These are added to the unit costs which are also expressed in per-tonne. (2) Also, why are the values zeroed?

1. Good catch, I have updated the comments
2. The values are zeroed in the defaults to the software because the transport costs are specific to the reference plant location and original tests were set up without the transport costs. I have added a test that uses non-zero feedstock transport costs and updated the feedstock transport costs in the reference plant input files.

RHammond2 commented on greenheart/simulation/technologies/steel/steel.py:
```diff
@@ -768,21 +786,21 @@ def run_steel_finance_model(
         usage=feedstocks.lime_consumption,
         unit="metric tonnes of lime per metric tonne of steel",
-        cost=feedstocks.lime_unitcost,
+        cost=(feedstocks.lime_unitcost + feedstocks.lime_transport_cost),
```
Just an FYI, that the parentheses aren't necessary from a purely code point of view, but are totally fine to leave.

RHammond2 reviewed (approved):
I left a comment that's just an FYI, so this looks good to go from my perspective!
